AdventOfCode2021/day3_2/day3_2.py

```diff
@@ -0,0 +1,74 @@
+#!python3
+
+from typing import List
+
+def file_read(file_name) -> List[str]:
+    file_data_list: List[str] = []
+    with open(file_name) as f:
+        for line in f:
+            file_data_list.append(line.strip())
+    return file_data_list
+
+
+def list_to_string(list_value: List[str]) -> str:
+    return ("".join([str(value) for value in list_value]))
+
+
+def list_of_binary_to_decimal(value: List[str]) -> int:
+    return int(list_to_string(value), 2)
+
+
+def most_common(bit0_numbers: List[str], bit1_numbers: List[str]) -> List[str]:
+    if (len(bit1_numbers) >= len(bit0_numbers)):
+        return bit1_numbers
+    else:
+        return bit0_numbers
+    
+    
+def least_common(bit0_numbers: List[str], bit1_numbers: List[str]) -> List[str]:
+    if (len(bit1_numbers) >= len(bit0_numbers)):
+        return bit0_numbers
+    else:
+        return bit1_numbers
+    
+
+def rating_value(data_list_original: List[str], fun) -> int:
+    data_list: List[str] = data_list_original
+    index_n_max = len(data_list_original[0]) - 1
+    index_n = 0
+    bit0_numbers: List[str] = []
+    bit1_numbers: List[str] = []
+
+    while (len(data_list) > 1):
+        for i in range(0, len(data_list)):
+            if data_list[i][index_n] == '1':
+                bit1_numbers.append(data_list[i])
+            elif data_list[i][index_n] == '0':
+                bit0_numbers.append(data_list[i])
+        data_list = fun(bit0_numbers, bit1_numbers)
+        bit1_numbers = []
+        bit0_numbers = []
+        index_n += 1
+        if (index_n > index_n_max): 
+            break   
+    
+    rating_integer_value: int = list_of_binary_to_decimal(data_list)
+    return rating_integer_value
+
+
+def main():
+    data_list_original: List[str] = file_read("day3_2.txt")
+    oxygen_generator_rating: int = 0
+    CO2_scrubber_rating: int = 0
```

Comment:
Let's call this variable `co2_scrubber_rating`, since python convention is to use lower case for variables names and upper case for classes names.

Reply:
got it!